code/datums/actions/item_action.dm

```diff
@@ -117,7 +117,6 @@
 	RegisterSignal(holder_flamer, COMSIG_ITEM_HYDRO_CANNON_TOGGLED, .proc/update_toggle_button_icon)
 
 /datum/action/item_action/toggle_hydro/action_activate()
-	. = ..()
 	holder_flamer.unique_action(owner)
```

Comment:
Explain this

Reply:
It was calling the above proc which ended up making the action happen twice.

Reply:
What above proc
Why didnt you just delete the unique action call

Reply:
I tried that and while it worked for making it not double, it didn't update the hud icon.

Reply:
This proc should be calling parent look into why its being called twice instead

Reply:
it uses ui_action_click()